### PR TITLE
RegistrySession: fail when creating for unknown registry

### PR DESCRIPTION
**Pull request by rcerven**
* CLOUDBLD-1196

Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [ ] Commit messages are descriptive enough
- [ ] Code coverage from testing does not decrease and new code is covered
- [ ] JSON/YAML configuration changes are updated in the relevant schema
- [ ] Changes to metadata also update the documentation for the metadata
- [ ] Pull request has a link to an osbs-docs PR for user documentation updates
- [ ] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/util.py [hunk 1]
```diff
@@ -670,12 +673,11 @@ def create_from_config(cls, workflow, registry=None, access=None):
                 raise RuntimeError(msg)
         else:
             # Try to match a registry in pull_registries, if no match found, use source_registry
```
Comment (chmeliik) [inline atomic_reactor/util.py:675]: Comment will need to be updated (possibly docstring as well)

File: tests/plugins/test_pin_operator_digests.py [hunk 2]
```diff
@@ -402,8 +403,8 @@ def test_orchestrator_exclude_csvs(self, docker_tasker, tmpdir, caplog, has_envs
     def test_orchestrator(self, docker_tasker, tmpdir, caplog):
         pullspecs = [
             # final-registry: do not replace registry or repos
-            'final-registry/ns/foo@sha256:1',  # -> no change
-            'final-registry/ns/foo:1',  # -> final-registry/ns/foo@sha256:1
+            'registry.private.example.com/ns/foo@sha256:1',  # -> no change
+            'registry.private.example.com/ns/foo:1',  # -> final-registry/ns/foo@sha256:1
```
Comment (chmeliik) [inline tests/plugins/test_pin_operator_digests.py:407]: `final-registry` still mentioned in comments

File: atomic_reactor/util.py (hunk 1 above)
Comment (rcerven) [reply atomic_reactor/util.py]: updated

File: tests/plugins/test_pin_operator_digests.py (hunk 2 above)
Comment (rcerven) [reply tests/plugins/test_pin_operator_digests.py]: renamed

**Comment by rcerven** [comment]
rn updated: RegistrySession: fail when creating for unknown registry

**Pull request merged**